azurerm/internal/services/network/application_gateway_resource.go

```diff
@@ -2594,6 +2601,7 @@ func expandApplicationGatewayProbes(d *schema.ResourceData) *[]network.Applicati
 				MinServers:                          utils.Int32(minServers),
 				Path:                                utils.String(probePath),
 				Protocol:                            network.ApplicationGatewayProtocol(protocol),
+				Port:                                utils.Int32(port),
```

Comment:
since this is an optional field, presumably a default value of `0` is invalid? if so, we may need to conditionally set this field? e.g.

```
if port != 0 {
  output. ApplicationGatewayProbePropertiesFormat.Port = utils.Int32(port)
}
```

Reply:
Makes sense. Fixed.